ArticleTemplates/assets/js/bootstraps/gallery.js

```diff
@@ -1,120 +1,114 @@
-import { debounce } from 'modules/util';
-
-let galleryProps;
-let galleryRows;
-
-const PROPS = {
-        0 :{
-            imagesPerRow: 3
-        },
-        650: {
-            imagesPerRow: 4
-        },
-        1200: {
-            imagesPerRow: 5
-        }
-    };
+import Hammer from 'hammerjs';
 
 function init() {
-    buildGallery();
-    window.addEventListener('resize', debounce(buildGallery, 100));
-}
+    lazyLoadImages();
+    var galleryImages = document.querySelectorAll('.touch-gallery__images');
+    galleryImages.forEach(function(galleryImage) {
+        var hammerSettings = (GU.opts.platform === 'ios' ? { touchAction: 'auto' } : {})
+        var mc = new Hammer.Manager(galleryImage, hammerSettings);
+        var pinch = new Hammer.Pinch();
+
+        mc.add(pinch);
+
+        mc.on('pinch', function(ev) {
+            if (GU.opts.platform === 'ios') {
+                ev.preventDefault();
+            }
 
-function buildGallery() {
-    setGalleryProps();
-    addImagesToRows();
-    addFlexClass();
-}
+            var desiredScale = ev.scale;
+            var pinchCentre = getPinchCentre(galleryImage);
 
-function setGalleryProps() {
-    let width;
+            galleryImage.style.transformOrigin = pinchCentre.x+'% '+pinchCentre.y+'%';
+            galleryImage.style.transform = 'scale('+desiredScale+')';
+        });
 
-    for (width in PROPS) {
-        if(PROPS.hasOwnProperty(width)) {
-            if (window.innerWidth > parseInt(width, 10)) {
-                galleryProps = PROPS[width];
-            } else {
-                break;
-            }
-        }
-    }
-}
+        mc.on('pinchstart', function(ev) {
+            galleryImage.classList.add('touch-gallery__images--pinch');
+            lockArticleSwipe(true);
+            savePinchCentre(galleryImage, ev);
+        });
 
-function addImagesToRows() {
-    let i;
-    const images = document.body.getElementsByClassName('gallery__image-container');
+        mc.on('pinchend', function() {
+            lockArticleSwipe(false);
+            bounceToInitialPosition(galleryImage);
+        });
 
-    const rows = [
-        []
-    ];
+    });
 
-    // loop through images and add to rows
-    for (i = images.length - 1; i >= 0; i--) {
-        // if current row is full create new row
-        if (rows[0].length === galleryProps.imagesPerRow) {
-            rows.unshift([]);
+    function lockArticleSwipe(toggle) {
+        var isAndroidApp = (window.location.origin === "file://" && /(android)/i.test(navigator.userAgent) ) ? true : false;
```

Comment:
Why are we testing if we're on the android app using this technique? Does `GU.opts.platform === 'android'` not behavew as you'd expect?

Reply:
Duarte had written that code before leaving so I'm not sure if there's a specific reason.
I've checked and `GU.opts.platform` returns "android" as expected so I've updated it.